pkg/apis/v1beta1/nodeclaim.go

```diff
@@ -32,6 +32,9 @@ type NodeClaimSpec struct {
 	// +optional
 	StartupTaints []v1.Taint `json:"startupTaints,omitempty"`
 	// Requirements are layered with GetLabels and applied to every node.
+	// +kubebuilder:validation:XValidation:message="requirements with operator 'In' must have a value defined",rule="self.all(x, x.operator == 'In' ? x.values.size() != 0 : true)"
+	// +kubebuilder:validation:XValidation:message="requirements operator 'Gt' or 'Lt' must have a single positive integer value",rule="self.all(x, (x.operator == 'Gt' || x.operator == 'Lt') ? x.values.size() == 1 && int(x.values[0]) >= 0: true)"
```

Comment:
```suggestion
	// +kubebuilder:validation:XValidation:message="requirements operator 'Gt' or 'Lt' must have a single positive integer value",rule="self.all(x, (x.operator == 'Gt' || x.operator == 'Lt') ? x.values.size() == 1 && int(x.values[0]) >= 0 : true)"
```

nit: add a space between the ternary operators. Maybe also consider throwing the `&&` statement in parenthesis for clarity